### PR TITLE
Fixed issue #17696: Mass changing expiration date fails (except for superadmin)

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/SurveyAdministrationController.php [hunk 1]
```diff
@@ -2632,7 +2615,7 @@ public function changeTemplate($iSurveyID, $template, $aResults = null, $bReturn
 
         $survey = Survey::model()->findByPk($iSurveyID);
 
-        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveyactivation', 'update')) {
+        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveysettings', 'update')) {
```
Comment (Shnoulle) [inline application/controllers/SurveyAdministrationController.php:2618]: User need read permission (usage) on template name too

File: application/controllers/SurveyAdministrationController.php [hunk 2]
```diff
@@ -2632,7 +2615,7 @@ public function changeTemplate($iSurveyID, $template, $aResults = null, $bReturn
...
             if (!empty($bReturn)) {
```
Comment (Shnoulle) [inline application/controllers/SurveyAdministrationController.php:2619]: See https://github.com/LimeSurvey/LimeSurvey/blob/0b3a1c0bcab8c5359595b027eafa2e0238318126/application/controllers/SurveyAdministrationController.php#L789 it's an empty array, then break if 1st survey don't have valid permission.

`!is_array` maybe ? Or isnull ?

**Review by Shnoulle — CHANGES_REQUESTED** [review]
See permission on new template (read) (maybe at start ?) 
See !empty for return.

File: application/controllers/SurveyAdministrationController.php (hunk 2 above)
Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:2619]: why? not following. Please notice this was already working... we only changed the permission checking...

Comment (Shnoulle) [reply application/controllers/SurveyAdministrationController.php:2619]: Yes but before user have all rights (it was superadmin) then permission return always true this part was never used .

`$aResults` start by an empty array, then if user don't have permission on the 1st surey : it die … https://github.com/LimeSurvey/LimeSurvey/blob/0b3a1c0bcab8c5359595b027eafa2e0238318126/application/controllers/SurveyAdministrationController.php#L2625

Comment (gabrieljenik) [reply application/controllers/SurveyAdministrationController.php:2619]: No, it is not die-ing :)
You can see `$bReturn` is sent as true sometimes, which returns the errors on an array.

**Review by Shnoulle — APPROVED** [review]
Code OK
Sorry for error about aResult != breturn …

**Pull request merged**